lib/commit-lint.ts

```diff
@@ -99,6 +100,14 @@ export class LintCommit {
         }
     }
 
+    // Reject commits that appear to require rebasing
+
+    private bangPrefix(): void {
+        if (this.lines[0].match(/^(squash|fixup|amend)!/)) {
+            this.block(`Rebase needed to squash commit: ${this.lines[0]}`);
```

Comment:
Every once in a while I send patch series with a `fixup!` for a buggy patch series. Could we turn this into a warning?

Reply:
I've tended to use `git format-patch` and paste my `fixup!` commits into a reply, giving me an opportunity to add extra context and keep the patch next to the discussion of the original topic. If that doesn't get the traction needed to apply the `fixup!`, then sending a forward fix as a non-`fixup!` is appropriate.

Would you consider changing your workflow a bit to allow us to help others avoid silly mistakes by keeping this as an error?

Reply:
Sure!